src/Sylius/Component/Core/OrderProcessing/PaymentProcessorInterface.php

```diff
@@ -24,4 +24,9 @@
      * @param OrderInterface $order
      */
     public function processOrderPayments(OrderInterface $order);
+
+    /**
+     * @param OrderInterface $order
+     */
+    public function createNewPaymentForOrder(OrderInterface $order);
```

Comment:
If it creates new payment for order there is no need for this method, because `processOrderPayments` handle this. The name is a little bit misleading. Moreover, IMHO this logic should be a part of `processOrderPayments`